cocos/base/CCDirector.cpp

```diff
@@ -210,6 +216,13 @@ Director::~Director(void)
     
     Configuration::destroyInstance();
 
+    Sprite3DMaterial::releaseBuiltInMaterial();
+
+    cocostudio::TransformHelp::release();
+
+    ObjectFactory::destroyInstance();
```

Comment:
I mean developers may remove cocos studio/particle3d from project building, so `Director` can not be so tightly with these modules.